### PR TITLE
Buildkite support

### PR DESCRIPTION
**Pull request by tetsuo-cpp**

### CODE REVIEW

**Comment by tetsuo-cpp** [comment]
@woodruffw @di I've run this on a BuildKite and can verify that it works against the staging instance.

File: id/_internal/oidc/ambient.py [hunk 1]
```diff
@@ -188,3 +190,55 @@ def detect_gcp(audience: str) -> Optional[str]:
...
+    # there doesn't seem to be any way to do this that satisfies `bandit` so I
+    # think we need to ignore this.
+    # More context at:
+    #   https://github.com/PyCQA/bandit/issues/333
+    #
+    # B607: This is complaining about invoking an external executable without
+    # providing an absolute path (we just refer to whatever `buildkite-agent`)
+    # is in the `PATH`. For a Buildkite agent, there's no guarantee where the
+    # `buildkite-agent` is installed so again, I don't think there's anything
+    # we can do about this.
```
Comment (woodruffw) [inline id/_internal/oidc/ambient.py:231]: Thanks for the detailed comment! Silencing these warnings looks good to me.

**Review by woodruffw — APPROVED** [review]
LGTM!

File: id/_internal/oidc/ambient.py [hunk 2]
```diff
@@ -188,3 +190,55 @@ def detect_gcp(audience: str) -> Optional[str]:
...
+    process = subprocess.run(  # nosec B603, B607
+        ["buildkite-agent", "oidc", "request-token", "--audience", "sigstore"],
```
Comment (di) [inline id/_internal/oidc/ambient.py:233]: The audience should be `audience` here, not `"sigstore"`.

Comment (woodruffw) [reply id/_internal/oidc/ambient.py:233]: Whoops, missed that. Fixing now.

Comment (woodruffw) [reply id/_internal/oidc/ambient.py:233]: https://github.com/di/id/pull/23

**Pull request merged**